services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -713,6 +718,62 @@ class FlowCrudSpec extends BaseSpecification {
         "an unmetered"  | 0
     }
 
+    @Unroll
+    @Tags(VIRTUAL)
+    def "Able to create an one-switch flow on a DEACTIVATED switch"(FlowPayload flow) {
+        given: "A deactivated switch"
+        def sw = topology.getActiveSwitches().find { it.dpId == flow.source.switchDpId }
+        lockKeeper.knockoutSwitch(sw)
+        TimeUnit.SECONDS.sleep(10) // for no good reason it fails without sleep
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        when: "Create a flow"
+        northbound.addFlow(flow)
+
+        then: "Flow is created with the 'Down' status"
+        TimeUnit.SECONDS.sleep(5)
+        Wrappers.wait(WAIT_OFFSET) {
+            northbound.getFlowStatus(flow.id).status == FlowState.DOWN
+        }
+
+        when: "Reconnect the switch back"
+        lockKeeper.reviveSwitch(sw)
+        TimeUnit.SECONDS.sleep(10)
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.ACTIVATED
+        }
+
+        then: "Flow status is changed to Up"
+        TimeUnit.SECONDS.sleep(5)
```

Comment:
?
Our tests should not just 'sleep', this is a bad design and only acceptable when you purely understand why the sleep is used. Otherwise - we have special 'wait' wrappers to make these sleeps as short as possible and clearly understand what we are waiting for